### PR TITLE
Only sort by group name when LDAP is involved

### LINKED ISSUE #16876: Deleting a user adds "NaN" where the user count of groups used to be
Regression from https://github.com/owncloud/core/pull/16402

### PR DESCRIPTION
**Pull request by nickvergessen**
Bringing back the group count, when LDAP is not involved: https://github.com/owncloud/core/pull/16402#issuecomment-111083373
Fix #16876

@blizzz @DeepDiver1975 @jancborchardt

### CODE REVIEW

**Comment by jancborchardt** [comment]
Hm, just tested and I can’t really see a difference to master. Strange

**Comment by jancborchardt** [comment]
(I definitely do not have LDAP enabled.)

File: settings/users.php [hunk 1]
```diff
@@ -37,12 +37,25 @@
 $userManager = \OC_User::getManager();
 $groupManager = \OC_Group::getManager();
 
+// Set the sort option: SORT_USERCOUNT or SORT_GROUPNAME
+$sortGroupsBy = \OC\Group\MetaData::SORT_USERCOUNT;
+$sortGroupsBy = \OC\Group\MetaData::SORT_GROUPNAME;
```
Comment (nickvergessen) [inline settings/users.php]: @jancborchardt well obvious fail is obvious

**Comment by nickvergessen** [comment]
@jancborchardt should be fixed

**Comment by jancborchardt** [comment]
Good stuff. :+1:

(Ideally the sidebar sorting and count updates live when groups are changed, but I think this didn’t work before either –> separate PR.)

**Comment by jancborchardt** [comment]
Another review please @Blizzz @DeepDiver1975 @MorrisJobke @owncloud/designers :)

File: lib/public/igroupmanager.php [hunk 2]
```diff
@@ -41,6 +41,12 @@
  */
 interface IGroupManager {
 	/**
+	 * @return \OC_Group_Backend[] Get registered backends
```
Comment (DeepDiver1975) [inline lib/public/igroupmanager.php]: hmmmm ... we shall not return private classes

Comment (DeepDiver1975) [inline lib/public/igroupmanager.php]: What about adding a method like

```php
/**
 * string $backendClass
 */
public function isBackendEnabled($backendClass);

Comment (nickvergessen) [inline lib/public/igroupmanager.php]: Fixed

**Comment by SergioBertolinSG** [comment]
Tested #16954 here, working fine.

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by jancborchardt** [comment]
So we got a +1 from @DeepDiver1975 or @SergioBertolinSG?

**Comment by blizzz** [comment]
Works as intended with and without LDAP. :+1:

**Comment by blizzz** [comment]
@owncloud-bot retest this please

**Pull request merged**